src/megaapi_impl.cpp

```diff
@@ -25184,14 +25304,73 @@ void MegaFolderDownloadController::cancel()
     //remove subtransfers from pending transferQueue
     megaApi->cancelPendingTransfersByFolderTag(tag);
 
+    //remove ongoing subtransfers
+    long long cancelledSubTransfers = 0;
+    DBTableTransactionCommitter committer(client->tctable);
     while (!subTransfers.empty())
     {
         auto subTransfer = *subTransfers.begin();
-        subTransfer->setState(MegaTransfer::STATE_COMPLETED);
-        DBTableTransactionCommitter committer(client->tctable);
-        megaApi->fireOnTransferFinish(subTransfer, make_unique<MegaErrorPrivate>(API_EINCOMPLETE), committer);
+
+        Transfer *transfer = subTransfer->getTransfer();
+        if (!transfer)
+        {
+            LOG_warn << "Subtransfer without attached Transfer for folder transfer: " << subTransfer->getFileName();
+
+            subTransfer->setState(MegaTransfer::STATE_COMPLETED); //TODO: copmleted?
```

Comment:
Not sure about the implications of the state of the transfer. Isn't it exclusively for the app? Maybe it's more appropriate to report it as `STATE_CANCELLED`.
(same here:
https://github.com/meganz/sdk/pull/2087/files#diff-c3de3d92d9d9472cc3d715cb978d2940R23952
https://github.com/meganz/sdk/pull/2087/files#diff-c3de3d92d9d9472cc3d715cb978d2940R23905
https://github.com/meganz/sdk/pull/2087/files#diff-c3de3d92d9d9472cc3d715cb978d2940R25366)